src/mumble/GlobalShortcut_win.cpp

```diff
@@ -208,6 +208,103 @@ void GlobalShortcutWin::registerMetaTypes() {
 	}
 }
 
+QList< Shortcut > GlobalShortcutWin::migrateSettings(const QList< Shortcut > &oldShortcuts) {
+	constexpr QUuid keyboardUuid(0x6F1D2B61, 0xD5A0, 0x11CF, 0xBF, 0xC7, 0x44, 0x45, 0x53, 0x54, 0x00, 0x00);
+	constexpr QUuid mouseUuid(0x6F1D2B60, 0xD5A0, 0x11CF, 0xBF, 0xC7, 0x44, 0x45, 0x53, 0x54, 0x00, 0x00);
+	constexpr QUuid xinputUuid(0xCA3937E3, 0x640C, 0x4D9E, 0x9E, 0xF3, 0x90, 0x3F, 0x8B, 0x4F, 0xBC, 0xAB);
+	constexpr QUuid gkeyKeyboardUuid(0x153E64E6, 0x98C8, 0x4E, 0x03, 0x80EF, 0x5F, 0xFD, 0x33, 0xD2, 0x5B, 0x8A);
+	constexpr QUuid gkeyMouseUuid(0xC41E60AF, 0x9022, 0x46CF, 0xBC, 0x39, 0x37, 0x98, 0x10, 0x82, 0xD7, 0x16);
+
+	QList< Shortcut > newShortcuts;
+
+	for (auto shortcut : oldShortcuts) {
```

Comment:
Imo `auto` should not be used for this. Also: Can the loop variable be `const`?

Reply:
No, we are modifying a part of the data it contains.

Reply:
I would keep `auto` because the datatype (`Shortcut`) is in the function signature.

Reply:
The whole point is not that one can't figure out the type of the variable but that one has to switch to a different part of the code in order to do so which makes reading the code much more tedious than it needs to be.

Asking differently: What's the advantage of using `auto` here instead of directly writing down the type?

Reply:
It's shorter and the datatype is implied based on the variable's name.

I can understand avoiding `auto` when the datatype is far away, but in this case it's just above and `newShortcuts` is the same as `oldShortcuts`.

Reply:
See our conclusion of our meeting from yesterday :)